README.md
# appDevelop
study react-native


CHANGELOG.md
更新日志
---

## 0.2.0 (2017-07-04)

Feature:

- 登录页面
- 登录页面接口请求及token存储


## 0.1.0 (2017-06-25)

Feature:

- App页面编写

src/business/Cell.js
import React, { Component } from 'react';
import { View, Text, Image, StyleSheet } from "react-native";

const styles = StyleSheet.create({
  cell: {
    width: '94%',
    height: 110,
    backgroundColor: "#fff",
    borderWidth: 1,
    borderStyle: "solid",
    borderColor: "transparent",
    borderRadius: 10,
    marginTop: 16,
    flexDirection: "row"
  },
  head: {
    width: "30%",
    height: "100%",
    justifyContent: "center",
    alignItems: 'flex-end'
  },
  content: {
    width: "50%",
    height: "100%",
    justifyContent: "center",
    alignItems: 'flex-start',
    paddingLeft:10
  },
  next: {
    width: "20%",
    height: "100%",
    justifyContent: "center",
    alignItems:"center"
  },
  odd:{
    width: 70,
    height: 70,
    borderWidth:1,
    borderStyle:"solid",
    borderColor:"transparent",
    borderRadius:35,
    justifyContent: "center",
    alignItems: 'center',
    backgroundColor: "#80cafb"
  },
  even: {
    width: 70,
    height: 70,
    borderWidth:1,
    borderStyle:"solid",
    borderColor:"transparent",
    borderRadius:35,
    justifyContent: "center",
    alignItems: 'center',
    backgroundColor: "#f9a686"
  }
});
const atm = require("./../images/atm.png");
const cattle = require("./../images/niu.png");

class Cell extends Component {
  render() {
    const { count, title, parity } = this.props;
    return (
      <View style={styles.cell}>
        <View style={styles.head}>
          <View style={parity === "odd" ? styles.odd : styles.even}>
            <Image source={parity === "odd" ? atm : cattle} style={{width:40,height:40}}/>
          </View>
        </View>
        <View style={styles.content}>
          <Text style={{fontSize: 18,fontWeight: "bold",lineHeight:40}}>{title}</Text>
          <Text style={{color: "#999", lineHeight:30}}>人数 {count}</Text>
        </View>
        <View style={styles.next}>
          <Image source={require("./../images/right.png")} style={{width:20,height:30}}/>
        </View>
      </View>
    )
  }
}

export default Cell;

src/business/BusinessDetail.js
import React from "react";
import { View, Text, Image, StyleSheet,TouchableOpacity  } from "react-native";
import Cell from "./Cell";
import ApiUtil from "./../util/ApiUtil";
import Storage from "./../util/Storage";

const styles = StyleSheet.create({
  view:{
    flex: 1,
    flexDirection: "column",
    backgroundColor: '#fff'
  },
  container: {
    flex: 1,
    alignItems: 'center',
    backgroundColor: '#edf1fc'
  },
  titleView: {
    height: 80,
    width: "100%",
    backgroundColor:"#363f5e"
  },
  title: {
    color: "#fff",
    fontSize: 18,
    fontWeight:"bold",
    textAlign: "center",
    marginTop:46
  },
  footer: {
    height: 60,
    flexDirection: "row",
    paddingLeft:30,
    paddingRight:30
  },
  footerContent: {
    width: "50%",
    alignItems: 'center',
    justifyContent: "center",
    flexDirection: "column"
  }
});

class BusinessDetail extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      activeKey: 1,
      token: null,
      data: {}
    }
  }

  componentDidMount() {
    Storage.get("token", token => {
      if(token) {
        ApiUtil.request("http://apicrm.nongfenqi.net/index/count", {}, "GET", false, token)
          .then(result => {
            this.setState({data: result.data || {}})
          })
      }
    });
  }

  render() {
    const { activeKey, data } = this.state;
    const activePeople = require("./../images/activePeople.png");
    const people = require("./../images/people.png");
    return (
      <View style={styles.view}>
        <View style={styles.container}>
          <View style={styles.titleView}>
            <Text style={styles.title}>工作台</Text>
          </View>
          <Cell count={data.customerCount} title="客户管理" parity="odd"/>
          <Cell count={data.applyOrderCount} title="业务管理" parity="even"/>
        </View>
        <View style={styles.footer}>
          <View style={styles.footerContent}>
            <TouchableOpacity  onPress={()=> {this.setState({activeKey: 1})}}>
              <Image source={ activeKey === 1 ? activePeople : people} style={{width:26,height:26}} />
            </TouchableOpacity >
            <Text style={{fontSize: 12, marginTop: 6, color: activeKey === 1 ? "#619af8" : "#2c2c2c"}}>工作区</Text>
          </View>
          <View style={styles.footerContent}>
            <TouchableOpacity  onPress={()=> {this.setState({activeKey: 2})}}>
              <Image source={ activeKey !== 1 ? activePeople : people} style={{width:26,height:26}} />
            </TouchableOpacity >
            <Text style={{fontSize: 12, marginTop: 6, color: activeKey !== 1 ? "#619af8" : "#2c2c2c"}}>我的</Text>
          </View>
        </View>
      </View>
    )
  }
}

export default BusinessDetail;


android/app/src/main/java/com/rnexample/MainApplication.java
package com.rnexample;

import android.app.Application;

import com.facebook.react.ReactApplication;
import com.facebook.react.ReactNativeHost;
import com.facebook.react.ReactPackage;
import com.facebook.react.shell.MainReactPackage;
import com.facebook.soloader.SoLoader;

import java.util.Arrays;
import java.util.List;

public class MainApplication extends Application implements ReactApplication {

  private final ReactNativeHost mReactNativeHost = new ReactNativeHost(this) {
    @Override
    public boolean getUseDeveloperSupport() {
      return BuildConfig.DEBUG;
    }

    @Override
    protected List<ReactPackage> getPackages() {
      return Arrays.<ReactPackage>asList(
          new MainReactPackage()
      );
    }
  };

  @Override
  public ReactNativeHost getReactNativeHost() {
    return mReactNativeHost;
  }

  @Override
  public void onCreate() {
    super.onCreate();
    SoLoader.init(this, /* native exopackage */ false);
  }
}


src/Login.js
import React from "react";
import { View, Text, TextInput, Image, Button, StyleSheet, AsyncStorage } from "react-native";
import Storage from "./util/Storage";
import ApiUtil from "./util/ApiUtil";

const styles = StyleSheet.create({
  view: {
    flex: 1,
    flexDirection: "column",
    backgroundColor: '#363f5e'
  },
  container: {
    flex: 1,
    alignItems: 'center',
    paddingLeft: 50,
    paddingRight: 50
  },
  userName: {
    width: 100,
    height: 100,
    backgroundColor: "#559aff",
    padding: 20,
    borderWidth: 1,
    borderStyle: "solid",
    borderColor: "transparent",
    borderRadius:51,
    justifyContent: "center",
    alignItems: 'center',
  },
  userNameDec: {
    color: "#fff",
    fontSize: 16,
    fontWeight: "bold"
  },
  innerUserView: {
    marginTop: 30,
    height: "38%",
    justifyContent: "center"
  },
  bossImage: {
    width:30,
    height:30
  },
  bossInput: {
    flex:1,
    marginLeft:10,
    height: 40,
    color: "#606f98"
  },
  bossButton: {
    width: "100%",
    height: 40,
    borderWidth: 1,
    borderStyle: "solid",
    borderColor: "transparent",
    backgroundColor: "#559aff",
    borderRadius: 5,
    marginTop: 40,
    paddingLeft: 5,
    paddingRight: 5
  },
  inputContainer: {
    width: "100%",
    borderColor: "#606f98",
    borderBottomWidth: 1,
    flexDirection:'row',
    marginTop: 12,
  },
  help: {
    width: "100%",
    textAlign: "center",
    color: "#5192f0"
  }
});

/**
 * 登录页面
 */

class Login extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      userName: null,
      pwd: null,
    };

    this.changePwd = this.changePwd.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
    this.changeUserName = this.changeUserName.bind(this);
  }

  changeUserName(value) {
    this.setState({userName: value})
  }

  changePwd(value) {
    this.setState({pwd: value})
  }

  handleSubmit() {
    const {pwd, userName} = this.state;
    if(!pwd || !userName) {
      alert("用户名或密码不能为空");
      return;
    }

    const params = {
      crmDevice: {
        deviceId: "63dff528-b645-449f-bc84-a95a8222a6b4",
      },
      password: pwd,
      username: userName
    };

    ApiUtil.request("http://apicrm.nongfenqi.net/user/login", params, "POST", true)
      .then(result => {
        if(result.retCode === 0) {
          const data = result.data || {};
          Storage.save("token",data.token && data.token.accessToken);
          const { navigate } = this.props.navigation;
          navigate('BusinessDetail');

        } else {
          alert(result.retMsg);
        }
      })
      .catch((error) => {
        alert(error);
      });
  }

  render() {
    return (
      <View style={styles.view}>
        <View style={styles.container}>
          <View style={styles.innerUserView}>
            <View style={styles.userName}>
              <Text style={styles.userNameDec}>Demon</Text>
            </View>
          </View>
          <View style={styles.inputContainer}>
            <Image source={require("./images/head.png")} style={styles.bossImage}/>
            <TextInput onChangeText={this.changeUserName} autoCorrect={false} autoCapitalize="none" style={styles.bossInput} placeholder="请输入手机号" placeholderTextColor="#606f98"/>
          </View>
          <View style={styles.inputContainer}>
            <Image source={require("./images/password.png")} style={styles.bossImage}/>
            <TextInput onChangeText={this.changePwd} secureTextEntry autoCapitalize="none" style={styles.bossInput} placeholder="请输入密码" placeholderTextColor="#606f98"/>
          </View>
          <View style={styles.bossButton}>
            <Button
              title="登 陆"
              color="#fff"
              onPress={this.handleSubmit}
            />
          </View>
        </View>
        <View style={{height: 60}}>
          <Text style={styles.help}>若忘记密码请拨打400-19001</Text>
        </View>
      </View>
    )
  }
}

export default Login;